src/Application/Actions/CreatePasswordResetToken.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace BigGive\Identity\Application\Actions;
+
+use BigGive\Identity\Client\Mailer;
+use BigGive\Identity\Domain\PasswordResetToken;
+use BigGive\Identity\Repository\PasswordResetTokenRepository;
+use BigGive\Identity\Repository\PersonRepository;
+use Laminas\Diactoros\Response\JsonResponse;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Validator\Constraints\Email;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class CreatePasswordResetToken extends Action
+{
+
+    public function __construct(
+        LoggerInterface $logger,
+        private readonly PersonRepository $personRepository,
+        private readonly PasswordResetTokenRepository $tokenRepository,
+        private readonly ValidatorInterface $validator,
+        private readonly Mailer $mailer,
+    ) {
+        parent::__construct($logger);
+    }
+
+    protected function action(): Response
+    {
+        $email = json_decode($this->request->getBody(), true)['email_address'];
+        $violations = $this->validator->validate($email, constraints: new Email());
+        if (count($violations) > 0) {
+            // todo return nicer error response here
+            throw new \Exception('Invalid email');
+        }
+
+        $person = $this->personRepository->findPasswordEnabledPersonByEmailAddress($email);
+
+        if ($person === null) {
+            // do not let the client know that we didn't find the person.
+            return new JsonResponse([]);
+        }
+
+        $token = new PasswordResetToken($person->getId());
+
+        $this->mailer->sendEmail([
+            'templateKey' => 'password-reset-requested',
+            'recipientEmailAddress' => $person->email_address,
+            'resetLink' => 'https://example.com/' . $token->toBase58Secret(), // @todo work out proper link
```

Comment:
Would it be simpler to just pass a `resetToken` and let Mailer worry about building the frontend link?

Reply:
It would be simpler here but it would make mailer more complex. I don't know if there's an advantage to putting that into mailer.